src/v/cloud_storage/types.h

```diff
@@ -343,6 +343,8 @@ struct anomalies
     uint32_t num_discarded_missing_segments{0};
     uint32_t num_discarded_metadata_anomalies{0};
 
+    bool segment_existence_checked{false};
```

Comment:
This is informational for the person reviewing scrub results. So if this is true, we know that the segment existence checks were performed as part of that scrub. I added this mostly because as a result of this PR, the scrub may run but not check for segments (because inv. data was missing). This flag lets a user determine if that happened. I will add a comment.